### PR TITLE
[P50] Camera Movement

### PR DESCRIPTION
**Pull request by EpicPants90**
https://trello.com/c/WZi4cHr4
P50 Camera Movement :dancer:

### CODE REVIEW

File: Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/CameraMovement.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    public float _ZoomModifier = 1;
+    public float _HorizontalModifier = 1;
+    public float _VerticalModifier = 1;
+    float _radius = 100;
+
+	// Use this for initialization
+	void Start () 
+    {
+        //start half way zoomed
+        _radius = Mathf.Lerp(5,20,0.5f);
```
Comment (alasdairhurst) [inline Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/CameraMovement.cs]: This should be using _MaxZoom and _MinZoom right?

**Comment by alasdairhurst** [comment]
@EpicPants90 Looks good, but there's a few changes i would recommend. One is kind of a bug where the camera doesn't follow the player cube while it's moving. At the moment the camera  won't move unless you rotate it after moving the player any number of squares.

The second is that the fully zoomed out camera should really be over the top of the board rather than still focusing on the player. It should also be looking directly down, pretty much 2D-like and if possible, fit the board to the screen.
example:
image
Then zooming in would go towards the player tile in the same way it does now.

A final tweak would be to decrease the rotation speed in general but more so the further you move out. It goes crazy when fully zoomed out :p

I changed the given variables - min zoom is now 3 so you see the tile a bit better, max zoom is 85, but would be different given the feedback above, zoom speed is 6 so it's much smoother.

No idea what horizontal/vertical modifers do since i changed horizontal and nothing seemed to happen.

**Comment by EpicPants90** [comment]
For now I fixed the bug where you can't move player when zoomed out and got the camera following the player.

**Comment by alasdairhurst** [comment]
@EpicPants90 Were you going to fix the bug which stops movement after a certain number of time in this PR?

**Pull request merged**